OpenSim/Simulation/StatesTrajectory.h

```diff
@@ -221,6 +230,164 @@ class OSIMSIMULATION_API StatesTrajectory {
     const_iterator end() const { return m_states.cend(); }
     /// @}
 
+
+    /// @name Accessing states by time
+    /// @{
+    /** Get the index of the nearest state before the given time
+     * (possibly *at* the given time).
+     *
+     * The tolerance exists to handle imprecise
+     * times. For example, if the trajectory contains states with times 1.3
+     * and 1.5 and you ask for the index of the state before time 1.4, you'll
+     * normally get the state with time 1.3. However, if you set a tolerance of
+     * 0.15, then you'll get the state with time 1.5 (because it is the state
+     * just before 1.55 = 1.4 + 0.15).
```

Comment:
This seems wrong or at least and awkward use case. First, using a tolerance of 0.15 should not be advocated - its purpose is to deal with imprecision not to expand the range.   Second the nearest **before** should still be enforced such that if 1.3 and 1.5 are equidistant fro 1.4 then 1.3 should prevail. In the case that 1.4 is 1.400000001 then the tolerance should reflect expected/typical truncation error. I don't  think the tolerance should default to 0 because users will not know what a reasonable tolerance should be and are going to be. We should use SimTK::Eps or the like as the default tolerance.